apps/web/pages/auth/oauth2/authorize.tsx

```diff
@@ -0,0 +1,170 @@
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+
+import { APP_NAME } from "@calcom/lib/constants";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Avatar, Button, Select } from "@calcom/ui";
+import { Plus, Info } from "@calcom/ui/components/icon";
+
+import PageWrapper from "@components/PageWrapper";
+
+export default function Authorize() {
+  const { t } = useLocale();
+  const { status } = useSession();
+
+  const router = useRouter();
+  const { state, client_id, scope } = router.query;
+  const queryString = router.asPath.split("?")[1] || "";
+
+  const [selectedAccount, setSelectedAccount] = useState<{ value: string; label: string } | null>();
+  const scopes = scope ? scope.toString().split(",") : [];
+
+  const { data: client, isLoading: isLoadingGetClient } = trpc.viewer.oAuth.getClient.useQuery(
+    {
+      clientId: client_id as string,
+    },
+    {
+      enabled: router.isReady && status !== "loading",
+    }
+  );
+
+  const { data, isLoading: isLoadingProfiles } = trpc.viewer.teamsAndUserProfilesQuery.useQuery();
+
+  const generateAuthCodeMutation = trpc.viewer.oAuth.generateAuthCode.useMutation({
+    onSuccess: (data) => {
+      window.location.href = `${client?.redirectUri}?code=${data.authorizationCode}&state=${state}`;
+    },
+  });
+
+  const mappedProfiles = data
+    ? data
+        .filter((profile) => !profile.readOnly)
+        .map((profile) => ({
+          label: profile.name || profile.slug || "",
+          value: profile.slug || "",
+        }))
+    : [];
+
+  useEffect(() => {
+    if (mappedProfiles.length > 0) {
+      setSelectedAccount(mappedProfiles[0]);
+    }
+  }, [isLoadingProfiles]);
+
+  const isLoading = isLoadingGetClient || isLoadingProfiles || !router.isReady || status === "loading";
+
+  if (isLoading) {
+    return <></>;
+  }
+
+  if (status === "unauthenticated") {
+    const urlSearchParams = new URLSearchParams({
+      callbackUrl: `auth/oauth2/authorize?${queryString}`,
+    });
+    router.replace(`/auth/login?${urlSearchParams.toString()}`);
+    return <></>;
+  }
```

Comment:
I think this should live in a useEffect 

```suggestion
useEffect(() => {
  if (status === "unauthenticated") {
    const urlSearchParams = new URLSearchParams({
      callbackUrl: `auth/oauth2/authorize?${queryString}`,
    });
    router.replace(`/auth/login?${urlSearchParams.toString()}`);
  }
}, [status]);
```